pertpy/tools/_sccoda.py

```diff
@@ -0,0 +1,2054 @@
+
+        return intercept_df
+
+    def summary(self, data: AnnData | MuData, extended: bool = False, modality_key: str = "coda", *args, **kwargs):
+        """Printing method for the summary.
+
+        Args:
+            data (AnnData | MuData): AnnData object or MuData object.
+            extended (bool, optional): If True, return the extended summary with additional statistics. Defaults to False.
+            modality_key (str, optional): If data is a MuData object, specify which modality to use. Defaults to "coda".
+            args: Passed to az.summary
+            kwargs: Passed to az.summary
+
+        Returns:
+            Prints to console
+        """
+        if isinstance(data, MuData):
+            try:
+                sample_adata = data[modality_key]
+            except IndexError:
+                print("When data is a MuData object, modality_key must be specified!")
+                raise
+        if isinstance(data, AnnData):
+            sample_adata = data
+        # Get model and effect selection types
+        select_type = sample_adata.uns["scCODA_params"]["select_type"]
+        model_type = sample_adata.uns["scCODA_params"]["model_type"]
+
+        # If other than default values for e.g. confidence interval are specified,
+        # recalculate them for intercept and effect DataFrames
+        if args or kwargs:
+            if model_type == "tree_agg":
+                intercept_df, effect_df, node_df = self.summary_prepare(sample_adata, *args, **kwargs)  # type: ignore
+            else:
+                intercept_df, effect_df, _ = self.summary_prepare(sample_adata, *args, **kwargs)  # type: ignore
+        # otherwise, get pre-calculated DataFrames. Effect DataFrame is stitched together from varm
+        else:
+            intercept_df = sample_adata.varm["intercept_df"]
+            covariates = sample_adata.uns["scCODA_params"]["covariate_names"]
+            effect_dfs = [sample_adata.varm[f"effect_df_{cov}"] for cov in covariates]
+            effect_df = pd.concat(effect_dfs)
+            effect_df.index = pd.MultiIndex.from_product(
+                (covariates, sample_adata.var.index.tolist()), names=["Covariate", "Cell Type"]
+            )
+            if model_type == "tree_agg":
+                node_df = sample_adata.uns["scCODA_params"]["node_df"]
+
+        # Get number of samples, cell types
+        data_dims = sample_adata.X.shape
+
+        if not extended:
+            # Cut down DataFrames to relevant info
+            intercept_df = intercept_df.loc[:, ["Final Parameter", "Expected Sample"]]
+
+            if model_type == "tree_agg":
+                node_df = node_df.loc[:, ["Final Parameter", "Is credible"]]
+                effect_df = effect_df.loc[:, ["Effect", "Expected Sample", "log2-fold change"]]
+            else:
+                effect_df = effect_df.loc[:, ["Final Parameter", "Expected Sample", "log2-fold change"]]
+
+        # Print everything neatly
```

Comment:
Use some some rich coloring here maybe? Or a rich table or something?

Reply:
Also, can we use f strings for all of this?

Reply:
Thanks for your suggestions! I didn't change johannes's original implementation when I integrate. Now, I added some rich color and used rich table to print dataframe. It is now much tidier and prettier